ansible-deployment/dnsmasq/tasks/main.yaml

```diff
@@ -0,0 +1,32 @@
+---
+- name: Custom dnsmasq configeration
+  hosts: overcloud_controllers
+  become: yes
+  tasks:
+     - name: Copy dnsmasq.conf
+       copy:
+         src: dnsmasq.conf
+         dest: /var/lib/config-data/puppet-generated/neutron/etc/neutron/
+         mode: 0644
+
+     - name: Edit dhcp_agent.ini
+       lineinfile:
+         path: /var/lib/config-data/puppet-generated/neutron/etc/neutron/dhcp_agent.ini
+         line: dnsmasq_config_file=/etc/neutron/dnsmasq.conf
+
+     - name: Install ipxe-bootimgs package
+       yum:
+         name: ipxe-bootimgs
+         state: present
+
+     - name: Copy ipxe boot images to /var/lib/neutron/dhcp
+       copy: src=/usr/share/ipxe/ dest=/var/lib/neutron/dhcp mode=0644 remote_src=yes
```

Comment:
you can similarly install `syslinux` and then copy the files from `/usr/share/syslinux` instead of having those file checked in the repo and then copying it.

sample contents from one of our pxe servers (hope it has all the files you are looking for)

```
[naved@kzn-vbmi01res syslinux]$ pwd
/usr/share/syslinux
[naved@kzn-vbmi01res syslinux]$ ls
altmbr.bin    cpuidtest.c32  gptmbr.bin    ifcpu64.c32     isohdppx_c.bin      mboot.c32    pcitest.c32   sanboot.c32     vesamenu.c32
altmbr_c.bin  diag           gptmbr_c.bin  ifcpu.c32       isohdppx_f.bin      mbr.bin      pmload.c32    sdi.c32         vpdtest.c32
altmbr_f.bin  disk.c32       gptmbr_f.bin  ifplop.c32      isolinux.bin        mbr_c.bin    poweroff.com  sysdump.c32     whichsys.c32
cat.c32       dmitest.c32    gpxecmd.c32   int18.com       isolinux-debug.bin  mbr_f.bin    pwd.c32       syslinux64.exe  zzjson.c32
chain.c32     dosutil        gpxelinux.0   isohdpfx.bin    kbdmap.c32          memdisk      pxechain.com  syslinux.com
cmd.c32       elf.c32        gpxelinuxk.0  isohdpfx_c.bin  linux.c32           memdump.com  pxelinux.0    syslinux.exe
config.c32    ethersel.c32   hdt.c32       isohdpfx_f.bin  ls.c32              meminfo.c32  reboot.c32    ver.com
cpuid.c32     gfxboot.c32    host.c32      isohdppx.bin    lua.c32             menu.c32     rosh.c32      vesainfo.c32
```

Reply:
Gotcha! I'll take a try.

Reply:
Done!